src/main/scala/chisel3/util/Counter.scala

```diff
@@ -59,6 +91,11 @@ object Counter
   @chiselName
   def apply(cond: Bool, n: Int): (UInt, Bool) = {
     val c = new Counter(n)
+    // Note this use of var is generally frowned upon! Don't use this as an example.
+    // This is done because we wanted the hardware generated by c.inc() to be wrapped
+    // in a when statement, but still needed to refer to the final value returned by
+    // c.inc() so we could return cond && wrap. Unless you really, really know what
+    // you are doing, IGNORE THIS CODE!!!!!
     var wrap: Bool = null
```

Comment:
I'm curious as to how this code avoids a NullPointerException...

Reply:
It always assigns to `wrap` the return value of `c.inc()`